tests/devapps/Intune-xamarin-ios/MainViewController.cs

```diff
@@ -0,0 +1,259 @@
+//-----------------------------------------------------------------------
+// <copyright file="MainViewController.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using CoreGraphics;
+using Foundation;
+using System;
+using System.IO;
+using System.Linq;
+using UIKit;
+using Microsoft.Intune.MAM;
+using Microsoft.Identity.Client;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace IntuneMAMSampleiOS
+{
+    public partial class MainViewController: UIViewController
+	{
+        UIButton btnMSAL;
+        UISwitch dirSwitch;
+
+        internal static IPublicClientApplication PCA { get; set; }
+        MainIntuneMAMComplianceDelegate _mamComplianceDelegate;
+        ManualResetEvent _manualReset;
+
+        public MainViewController(IntPtr handle) : base(handle)
+        {
+        }
+
+        public override void ViewDidLoad()
+        {
+            _manualReset = new ManualResetEvent(false);
```

Comment:
move the manual reset intuialiation inside the `MainIntuneMAMComplianceDelegate` ?